pangeo_forge_orchestrator/meta_types/meta.py

```diff
@@ -0,0 +1,59 @@
+from dataclasses import dataclass
+from typing import List, Literal, Optional
+
+
+@dataclass
+class Recipe:
+    id: Optional[str] = None
+    object: Optional[str] = None
+    dict_object: Optional[str] = None
+
+    def __post_init__(self):
+        if self.id is None and self.dict_object is None:
+            raise TypeError("Value needed for either id or dict_object")
+
+
+@dataclass
```

Comment:
```suggestion
@dataclass(frozen=True)
```

In general we should try to make as many of the types as possible frozen. Mutability = bugs.

Reply:
✅  https://github.com/pangeo-forge/pangeo-forge-orchestrator/pull/1/commits/68f534bbf02aeea0aa264e604df7b033703511a0